### PR TITLE
`BigInt` fixes corresponding to algebra#372

### PR DESCRIPTION
**Pull request by mmagician**
## Description

Companion to algebra's PR 372: https://github.com/arkworks-rs/algebra/pull/372

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (master)
- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests
- [ ] Updated relevant documentation in the code
- [ ] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: bls12_377/src/fields/tests.rs [hunk 1]
```diff
@@ -24,10 +24,7 @@ generate_field_serialization_test!(bls12_377; fq2; fq6; fq12;);
 
 #[test]
 fn test_fq_repr_from() {
-    assert_eq!(
-        BigInteger384::from(100),
-        BigInteger384([100, 0, 0, 0, 0, 0])
-    );
+    assert_eq!(BigInteger384::from(100), BigInt::new([100, 0, 0, 0, 0, 0]));
```
Comment (Pratyush) [inline bls12_377/src/fields/tests.rs:27]: I have an upcoming PR that adds `From` impls from u32, u16, u8, so I would change this to 

```suggestion
    assert_eq!(BigInteger384::from(100u64), BigInt::new([100, 0, 0, 0, 0, 0]));
```

File: bls12_377/src/fields/tests.rs [hunk 2]
```diff
@@ -46,7 +43,7 @@ fn test_fq_repr_is_odd() {
 fn test_fq_repr_is_zero() {
     assert!(BigInteger384::from(0).is_zero());
     assert!(!BigInteger384::from(1).is_zero());
```
Comment (Pratyush) [inline bls12_377/src/fields/tests.rs:45]:
```suggestion
    assert!(BigInteger384::from(0u8).is_zero());
    assert!(!BigInteger384::from(1u8).is_zero());
```

File: bls12_381/src/fields/tests.rs [hunk 3]
```diff
@@ -751,17 +751,14 @@ fn test_neg_one() {
...
     assert_eq!(negative_one, o);
 }
 
 #[test]
 fn test_fq_repr_from() {
-    assert_eq!(
-        BigInteger384::from(100),
-        BigInteger384([100, 0, 0, 0, 0, 0])
-    );
+    assert_eq!(BigInteger384::from(100), BigInt::new([100, 0, 0, 0, 0, 0]));
```
Comment (Pratyush) [inline bls12_381/src/fields/tests.rs:761]:
```suggestion
    assert_eq!(BigInteger384::from(100u64), BigInt::new([100, 0, 0, 0, 0, 0]));
```

File: bls12_381/src/fields/tests.rs [hunk 4]
```diff
@@ -780,12 +777,12 @@ fn test_fq_repr_is_odd() {
 fn test_fq_repr_is_zero() {
     assert!(BigInteger384::from(0).is_zero());
     assert!(!BigInteger384::from(1).is_zero());
```
Comment (Pratyush) [inline bls12_381/src/fields/tests.rs:779]:
```suggestion
    assert!(BigInteger384::from(0u64).is_zero());
    assert!(!BigInteger384::from(1u64).is_zero());
```

**Review by Pratyush — APPROVED** [review]

**Comment by weikengchen** [comment]
Finger crossed on the nightly :)

**Pull request merged**